### PR TITLE
Added DockerCompose file

### PR DESCRIPTION
**Pull request by mitri-slory**
The purpose of adding a Docker Compose file is to allow current and future backend engineers an easy set up for a local development environment. The issue at the moment is a ConnectionError with the ElasticSearch dependency for the devsetup container. With this issue, the root causes might be that the ES container hasn't finished setting up before the devsetup container's command runs or there might be a typo in the ES credentials in the config file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,9 +62,11 @@ All services share a single entry point in `main.py` file. This script dynamical
...
+To set up a local environment there is a special process to initialize a database and search cluster which is the `DevelopmentSetupProcess`. However, it's recommended to run the `DevelopmentSetupProcess` and `APIProcess` at the same time to build the most efficient local environment. To do so one of these commands should be run: `make up` or `docker compose up`. These commands will run the docker-compose file in the codebase so it's highly recommended to have Docker/Docker Desktop installed locally. After running one of the commands, a short import process will occur and populate the database with some sample data alongside running the API locally. This will allow you to query the API at `localhost:5000` and query the ESC at `localhost:9200`. 
 
-To run the API locally run `python main.py --process APIProcess` which will allow you to query the API at `localhost:5000`
+The docker compose file uses the sample-compose.yaml file in the `config` directory and additional configratuions and dependencies can be added to the file to build upon your local environment.
+
+To run the processes inividually the command should be in this format `python main.py --process APIProcess`.
```
Comment (mwbenowitz) [inline README.md:69]: Can we add a list of the environment variables that people will need to set in order to configure the app properly? Just the names of the HathiTrust and OCLC variables that should be set. We can add a note that they should either get their own or speak to another developer.

**Review by mwbenowitz — COMMENTED** [review]
Looks really good! Ready to be moved out of draft and should be able to be approved once with a few minor documentation additions

File: README.md [hunk 2]
```diff
@@ -62,9 +62,11 @@ All services share a single entry point in `main.py` file. This script dynamical
...
 - `--singleRecord` Accepts a single record identifier for the current process and imports that record only. Setting this will ignore `ingestType`, `limit` and `offset`.
 
-To set up a local environment there is a special process to initialize a database and search cluster. To set this up run `python main.py --process DevelopmentSetupProcess` which will run a short import process and populate the database with some sample data.
+To set up a local environment there is a special process to initialize a database and search cluster which is the `DevelopmentSetupProcess`. However, it's recommended to run the `DevelopmentSetupProcess` and `APIProcess` at the same time to build the most efficient local environment. To do so one of these commands should be run: `make up` or `docker compose up`. These commands will run the docker-compose file in the codebase so it's highly recommended to have Docker/Docker Desktop installed locally. After running one of the commands, a short import process will occur and populate the database with some sample data alongside running the API locally. This will allow you to query the API at `localhost:5000` and query the ESC at `localhost:9200`.
```
[the remark this answers is not in the document]
Comment (mitri-slory) [reply README.md:65]: I agree with changing the wording to "required." I'll put the required Docker/Docker Desktop line at the very beginning of the running services section to make it clear before engineers read the rest of the section.

File: scripts/deleteElasticRecords.py [hunk 3]
```diff
@@ -31,7 +31,7 @@ def main():
 
     dbManager.createSession()
 
-    batchSize = 10
+    batchSize = 1000
```
[the remark this answers is not in the document]
Comment (mitri-slory) [reply scripts/deleteElasticRecords.py:34]: I thought I already committed this change from a previous PR. I think I changed this to match what the actual batch size was when I ran this script before.

File: README.md (hunk 1 above)
Comment (mitri-slory) [reply README.md:69]: Ok I'll do that.

**Review by mwbenowitz — APPROVED** [review]
Looks great! This should be a big improvement in developer quality-of-life in the future

**Pull request merged**